src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -113,15 +113,13 @@ public async Task<IAccount> GetAccountAsync(string accountId)
         /// Removes all tokens in the cache for the specified account.
         /// </summary>
         /// <param name="account">Instance of the account that needs to be removed</param>
-        public Task RemoveAsync(IAccount account)
+        public async Task RemoveAsync(IAccount account)
         {
             RequestContext requestContext = CreateRequestContext(Guid.NewGuid());
-            if (account != null)
+            if (account != null && UserTokenCacheInternal != null)
```

Comment:
>UserTokenCacheInternal [](start = 35, length = 22)

silly question...what about conf clients that have an app cache and could also have a user cache?

Reply:
How do we remove tokens from the AppCache?

---
In reply to: [278221829](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1088#discussion_r278221829) [](ancestors = 278221829)

Reply:
Well, we've always only done this against the usertokencache.  this check just ensures we don't call Remove AccountAsync if the usertokencache is null (nullref exception).    agree that this doesn't help the app cache (good catch).  The public API has RemoveAsync only.  So should this (on confidential clients) look in both caches for this account?  Given that this PR isn't meant to change this functionality I'd argue to file a bug and we can look at how we want to fix this.